### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by Thovenfish** [magic282/NeuSum#28]
I got  "segmentation fault" , Could you tell me how can I sovle this problem please? Thanks~~

`2020-07-14 23:50:15,572 [INFO:onlinePreprocess]: WARNING: ignoring an empty line (21549)
2020-07-14 23:50:15,582 [INFO:onlinePreprocess]: WARNING: oracle exceeds max_doc_len, ignoring (21567)
2020-07-14 23:50:15,608 [INFO:onlinePreprocess]: WARNING: oracle exceeds max_doc_len, ignoring (21604)
2020-07-14 23:50:15,635 [INFO:onlinePreprocess]: WARNING: ignoring an empty line (21649)
2020-07-14 23:50:15,788 [INFO:onlinePreprocess]: WARNING: ignoring an empty line (21890)
2020-07-14 23:50:15,812 [INFO:onlinePreprocess]: WARNING: source and target do not have the same number of sentences
2020-07-14 23:50:15,813 [INFO:onlinePreprocess]: ... sorting sentences by size
2020-07-14 23:50:16,102 [INFO:onlinePreprocess]: Prepared 21924 sentences (0 ignored due to length == 0 or > 100)
2020-07-14 23:50:16,102 [INFO:__main__]:  * vocabulary size. source = 186521; target = 185305
2020-07-14 23:50:16,102 [INFO:__main__]:  * number of training sentences. 21924
2020-07-14 23:50:16,102 [INFO:__main__]:  * maximum batch size. 64
2020-07-14 23:50:16,102 [INFO:__main__]: Building model...
/home/shenyt/miniconda3/envs/sum_group1/lib/python3.6/site-packages/torch/nn/modules/rnn.py:54: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.3 and num_layers=1
  "num_layers={}".format(dropout, num_layers))
/home/shenyt/miniconda3/envs/sum_group1/lib/python3.6/site-packages/torch/nn/modules/rnn.py:54: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
  "num_layers={}".format(dropout, num_layers))
2020-07-14 23:50:16,447 [WARNING:__main__]: Not updating encoder word embedding.
2020-07-14 23:50:16,447 [INFO:__main__]: sent_encoder.word_lut.weight
/home/shenyt/program/neusum/code/NeuSum/neusum_pt/run.sh: line 55: 30469 Segmentation fault      (core dumped) python train.py -save_path ${SAVEPATH} -online_process_data -max_doc_len 80 -train_oracle ${DATAHOME}/train/out.txt -train_src ${DATAHOME}/train/train.txt.src -train_src_rouge ${DATAHOME}/train/out-rouge.txt -src_vocab ${DATAHOME}/train/train.src.vocab.txt -train_tgt ${DATAHOME}/train/train.txt.tgt -tgt_vocab ${DATAHOME}/train/train.tgt.vocab.txt -layers 1 -word_vec_size 50 -sent_enc_size 256 -doc_enc_size 256 -dec_rnn_size 256 -sent_brnn -doc_brnn -dec_init simple -att_vec_size 256 -norm_lambda 20 -sent_dropout 0.3 -doc_dropout 0.2 -dec_dropout 0 -batch_size 64 -beam_size 1 -epochs 100 -optim adam -learning_rate 0.001 -halve_lr_bad_count 100000 -gpus 0 -curriculum 0 -extra_shuffle -start_eval_batch 1000 -eval_per_batch 1000 -log_interval 100 -log_home ${SAVEPATH} -seed 12345 -cuda_seed 12345 -pre_word_vecs_enc ${DATAHOME}/glove/glove.6B.50d.txt -freeze_word_vecs_enc -dev_input_src ${DATAHOME}/dev/val.txt.src.shuffle.4k -dev_ref ${DATAHOME}/dev/val.txt.tgt.shuffle.4k -max_decode_step 3 -force_max_len`

**Comment by Thovenfish** [comment]
Through debugging, I find that when use the function which named "xavier_normal" the program will wrong. Could you tell me how can I sovle this problem please? Thanks~~

**Comment by magic282** [comment]
https://github.com/magic282/NeuSum#about-this-code
